File: src/main/java/com/example/web_chat/PresentationLayer/DTO/Outgoing/ChatMessageDTO.java
package com.example.web_chat.PresentationLayer.DTO.Outgoing;

public class ChatMessageDTO
{
    private Long id;

    private String roomName;

    private long creationTimestamp;

    private String text;

    // JSON deserialization requires a default constructor to be present
    public ChatMessageDTO(){}

    public ChatMessageDTO(long id, String roomName, long unixTimestamp, String text)
    {
        this.id = id;
        this.roomName = roomName;
        this.creationTimestamp = unixTimestamp;
        this.text = text;
    }

    public long getID()
    {
        return this.id;
    }

    public String getText()
    {
        return this.text;
    }

    public String getRoomName()
    {
        return this.roomName;
    }

    public long getCreationTimestamp()
    {
        return this.creationTimestamp;
    }
}

File: chat-app/app/src/DTOs/ChatMessageDTO.ts
export interface ChatMessageJSON 
{
  id: number;
  roomName: string;
  creationTimestamp: number;
  text: string;
}

export class ChatMessageDTO
{
    private id: number;
    private roomName: string;
    private creationTimestamp: number;
    private text: string;

    constructor(id: number, roomName: string, creationTimestamp: number, text: string)
    {
        this.id = id;
        this.roomName = roomName;
        this.creationTimestamp = creationTimestamp;
        this.text = text;
    }

    getID(): number
    {
        return this.id;
    }

    getText(): string
    {
        return this.text;
    }

    getRoomName(): string
    {
        return this.roomName;
    }

    getCreationTimestamp(): number
    {
        return this.creationTimestamp;
    }

    static fromJSON(json: ChatMessageJSON): ChatMessageDTO
    {
        const newChatMessageDTO = new ChatMessageDTO(json.id, json.roomName, json.creationTimestamp, json.text);
        return newChatMessageDTO;
    }
}
